src/libraries/System.Net.Http/src/System/Net/Http/ByteArrayContent.cs

```diff
@@ -46,6 +47,23 @@ public ByteArrayContent(byte[] content, int offset, int count)
             _count = count;
         }
 
+        protected override void SerializeToStream(Stream stream, TransportContext? context, CancellationToken cancellationToken)
+        {
+            // Only skip the original protected virtual SerializeToStream if this
+            // isn't a derived type that may have overridden the behavior.
+            if (GetType() != typeof(ByteArrayContent))
```

Comment:
We don't need to do this special-casing anymore.

The reason we do it in other places, like on Stream, is because the derived class may have already overridden an existing virtual method, and we need to make sure that if logically the new API performs the same task as the existing API and if callers might switch to use the new API instead of the current but still expect the same behavior, that the new API delegates to the existing one.  But with these new sync APIs, we're not delegating to any existing behavior: if the new method isn't overridden, we throw.  So it should be fine to just do the simple thing and have the implementation of this method be what's in SerializeToStreamCore.  That will then let you delete code from types derived from ByteArrayContent as well.